heimlig/src/hsm/keystore.rs

```diff
@@ -106,50 +134,117 @@ impl<const STORAGE_SIZE: usize, const NUM_KEYS: usize> KeyStore
     for MemoryKeyStore<STORAGE_SIZE, NUM_KEYS>
 {
     fn is_stored(&self, id: KeyId) -> bool {
-        match self.infos.iter().find(|key_info| key_info.id == id) {
+        match self
+            .layout
+            .iter()
+            .find(|key_layout| key_layout.info.id == id)
+        {
             None => false,
-            Some(key_info) => key_info.actual_size > 0,
+            Some(key_layout) => key_layout.actual_size > 0,
         }
     }
 
     fn size(&self, id: KeyId) -> Result<usize, Error> {
-        match self.infos.iter().find(|key_info| key_info.id == id) {
```

Comment:
Does it make sense to have some function that returns `Option<KeyLayout>` that can be used by `is_stored`, `export`, `size`, `delete` etc functions?

Reply:
Yes, there is `get_key_layout` now.